### DISCUSSION TITLE
`rye venv` command

### DISCUSSION THREAD

**Discussion by methane** [astral-sh/rye#626] [Ideas]
As a OSS maintainer, my most frequent workflow is creating minimum reproducible environment.
Current my workflow is:

```
$ mkdir issue1234
$ cd issue1234
$ rye pin 3.11  # global-python is enabled
$ mkvenv        # alias mkvenv='python3 -m venv --prompt . .venv && source .venv/bin/activate'
(issue1234) $ pip install <required packages>
(issue1234) $ cat >t.py  # copy&paste script to reproduce
(issue1234) $ python t.py
```

I use pip instead of `rye init && rye add && rye sync` because:

- I can see and fix build error quickly with `pip install -v`
- I want to use `pip install -U ../..` to test a branch containing bugfix
- I don't need `src/` directory

I am satisfied with this workflow. I think many OSS maintainers would like this workflow too.
So it would be nice if we can create .venv without some hacks (e.g. global-python and mkvenv alias).

- `rye venv --py 3.11` calls `$HOME/.rye/py/cpython@3.11.7/install/bin/python -m venv --prompt . .venv`
- `rye venv --py 3.11 VENV_DIR` calls `$HOME/.rye/py/cpython@3.11.7/install/bin/python -m venv VENV_DIR`

**Comment by mitsuhiko** [comment]
I do think there is some value in this, and I still use something like this myself. There is however also the risk of introducing yet another way of managing things and I was hoping to avoid this. Maybe it would make sense to have some "legacy plumbing" commands in Rye that enable something like this. I will need to think about this a bit more though.

**Comment by johnslavik** [comment]
Bump on this. It would definitely help in non-Rye projects where I want to use Rye because of how amazing it is.
As for now, I work around it by:
```bash
git add -A && git stash
rye pin 3.8  # my desired py version
# sometimes: rye init
rye sync
uv pip install ...
git add -A && git reset --hard HEAD && git stash pop
# now some empty dirs are left to manually delete...
```
(`.venv/` is never tracked by git due to my global git config)
If there was a `rye venv` command, I could simply use a dedicated command for my use case:
```
rye venv --py 3.8
uv pip install ...
```
Beware I don't want Rye's lock files, I just want it for simple toolchain & dependency management.

**Reply by johnslavik** [reply]
Funnily, I just found a better workaround that produces an unintended behavior Rye-wise.

```py
rye pin 3.8 && rye run uv venv
uv pip install ...
```

**Comment by God-damnit-all** [comment]
I ended up just making a rye command that opens a new shell as `rye run venv`

Honestly, I was really surprised to learn this didn't already exist.
